plugins/catalog-react/src/hooks/useEntityListProvider.tsx

```diff
@@ -214,20 +294,33 @@ export const EntityListProvider = <EntityFilters extends DefaultEntityFilters>(
         window.history?.replaceState(null, document.title, newUrl);
       }
     },
-    [catalogApi, queryParameters, requestedFilters, outputState],
+    [
+      catalogApi,
+      queryParameters,
+      requestedFilters,
+      outputState,
+      cursor,
+      enablePagination,
+    ],
     { loading: true },
   );
 
   // Slight debounce on the refresh, since (especially on page load) several
   // filters will be calling this in rapid succession.
-  useDebounce(refresh, 10, [requestedFilters]);
+  useDebounce(refresh, 10, [requestedFilters, cursor]);
 
   const updateFilters = useCallback(
     (
       update:
         | Partial<EntityFilter>
         | ((prevFilters: EntityFilters) => Partial<EntityFilters>),
     ) => {
+      // changing filters will affect pagination, so we need to reset
+      // the cursor and start from the first page.
+      // TODO(vinzscam): this is currently causing issues at page reload
+      // where the state is not kept. Unfortunately we need to rething
+      // the way filters work in order to fix this.
```

Comment:
So does this mean that reloading the page will lose track of the current cursor? Feels alright to ship but ofc nice to fix